airflow/executors/base_executor.py

```diff
@@ -288,7 +288,8 @@ def _emit_metrics(self, open_slots, num_running_tasks, num_queued_tasks):
         self.log.debug("%s running task instances for executor %s", num_running_tasks, name)
         self.log.debug("%s in queue for executor %s", num_queued_tasks, name)
         if open_slots == 0:
-            self.log.info("Executor parallelism limit reached. 0 open slots.")
+            if self.parallelism:
```

Comment:
FYI @BasPH

You changed the below log to info, and when parallelism is 0 this gets logged every single heartbeat because the open slots calculation in the sync method above doesn't account for that case very well. I didn't want to tackle that case just yet, because it would be kind of a breaking change for the metrics in that case (they would start reporting "infinity" open slots). So I went with not emitting that metrics if we see zero slots (nothing queued) and parallelism is zero.

Reply:
Good one :)